specs/lib/parseBinaryGltfSpec.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+var fs = require('fs');
+var async = require('async');
+var bufferEqual = require('buffer-equal');
+var parseBinaryGltf = require('../../lib/parseBinaryGltf');
+var binaryGltfPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest.glb';
+var gltfScenePath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTestBinary.txt';
+var fragmentShaderPath = './specs/data/boxTexturedUnoptimized/CesiumTexturedBoxTest0FS_Binary.glsl';
+var imagePath = './specs/data/boxTexturedUnoptimized/Cesium_Logo_Flat_Binary.png';
+
+describe('parseBinaryGltf', function() {
+    var testData = {
+        binary: binaryGltfPath,
+        gltf: gltfScenePath,
+        shader: fragmentShaderPath,
+        image: imagePath
+    };
+
+    beforeAll(function(done) {
+        var names = Object.keys(testData);
+        async.each(names, function(name, callback) {
+            fs.readFile(testData[name], function(err, data) {
+                if (err) {
+                    callback(err);
+                }
+                else {
+                    testData[name] = data;
+                    callback();
+                }
+            });
+        }, function(err) {
+            if (err) {
+                throw err;
+            }
+            done();
+        });
+    });
+
+    it('loads a glTF scene', function() {
+        var gltf = parseBinaryGltf(testData.binary);
+
+        expect(gltf).toBeDefined();
+        expect(gltf.buffers.KHR_binary_glTF).toBeDefined();
+        expect(gltf.images.Image0001).toBeDefined();
+        expect(gltf.shaders.CesiumTexturedBoxTest0FS).toBeDefined();
+        expect(gltf.shaders.CesiumTexturedBoxTest0VS).toBeDefined();
+        delete gltf.buffers.KHR_binary_glTF.extras;
+        delete gltf.images.Image0001.extras;
+        delete gltf.shaders.CesiumTexturedBoxTest0FS.extras;
+        delete gltf.shaders.CesiumTexturedBoxTest0VS.extras;
+
+        expect(gltf).toEqual(JSON.parse(testData.gltf));
+    });
+
+    it('loads an embedded image', function() {

```

Comment:
In a separate PR and once we have our binary glTF writer, let's add an "integration test" in the sense that we'll take our test glTF asset with all the properties that can be removed, convert it to binary glTF, then the remove optimizations, and then verify that it generated an optimized asset.